presto-main/src/test/java/com/facebook/presto/operator/aggregation/differentialentropy/TestWeightedReservoirAggregation.java

```diff
@@ -0,0 +1,121 @@
+import static org.testng.Assert.fail;
+
+public class TestWeightedReservoirAggregation
+        extends TestReservoirAggregation
+{
+    private InternalAggregationFunction function;
+
+    @BeforeClass
+    public void setUp()
+    {
+        FunctionManager functionManager = MetadataManager.createTestMetadataManager().getFunctionManager();
+        function = functionManager.getAggregateFunctionImplementation(
+                functionManager.lookupFunction(
+                        "differential_entropy",
+                        fromTypes(BIGINT, DOUBLE, DOUBLE)));
+    }
+
+    @Test
+    public void testBadNumberOfSamples()
+    {
+        try {
+            assertAggregation(
+                    this.function,
+                    0.0,
+                    createLongsBlock(Long.valueOf(-200)),
+                    createDoublesBlock(Double.valueOf(0.1)),
+                    createDoublesBlock(Double.valueOf(0.2)));
+            fail("Expected PrestoException");
+        }
+        catch (PrestoException e) {
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("sample"));
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("-200"));
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("positive"));
+        }
+    }
+
+    @Test
+    public void testNegativeWeight()
+    {
+        try {
+            assertAggregation(
+                    this.function,
+                    0.0,
+                    createLongsBlock(Long.valueOf(200)),
+                    createDoublesBlock(Double.valueOf(0.1)),
+                    createDoublesBlock(Double.valueOf(-0.2)));
+            fail("Expected PrestoException");
+        }
+        catch (PrestoException e) {
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("weight"));
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("negative"));
+            assertTrue(e.getMessage().toLowerCase(Locale.ENGLISH).contains("-0.2"));
+        }
+    }
+
+    @Override
+    public Block[] getSequenceBlocks(int start, int length)
+    {
+        int sequenceLength = 2 * length;
+        BlockBuilder samplesBuilder = DOUBLE.createBlockBuilder(null, sequenceLength);
+        BlockBuilder weightsBuilder = DOUBLE.createBlockBuilder(null, sequenceLength);
+        for (int weight = 1; weight < 3; ++weight) {
+            for (int i = start; i < start + length; i++) {
+                DOUBLE.writeDouble(samplesBuilder, Double.valueOf(i));
+                DOUBLE.writeDouble(weightsBuilder, Double.valueOf(weight));
+            }
+        }
+
+        return new Block[] {
+                createRLEBlock(TestReservoirAggregation.SIZE, sequenceLength),
+                 samplesBuilder.build(),
+                weightsBuilder.build()
+        };
+    }
+
+    @Override
+    protected List<String> getFunctionParameterTypes()
+    {
+        return ImmutableList.of(
```

Comment:
put the entire statement on the same line: `return ImmutableList.of(StandardTypes.INTEGER, StandardTypes.DOUBLE, StandardTypes.DOUBLE);`

Reply:
Did it in `TestUnweightedReservoirAggregation` too.